File: src/main/java/com/giffardtechnologies/restdocs/domain/type/BasicType.java
package com.giffardtechnologies.restdocs.domain.type;

public enum BasicType {
	INT, LONG, FLOAT, DOUBLE, STRING, BOOLEAN

	;
	public DataType asDataType() {
		switch (this) {
			case INT:
				return DataType.INT;
			case LONG:
				return DataType.LONG;
			case FLOAT:
				return DataType.FLOAT;
			case DOUBLE:
				return DataType.DOUBLE;
			case STRING:
				return DataType.STRING;
			case BOOLEAN:
				return DataType.BOOLEAN;
		}
		throw new UnsupportedOperationException("Could not map " + this + " to DataType");
	}
}

File: src/main/java/com/giffardtechnologies/restdocs/domain/type/BasicType.kt
package com.giffardtechnologies.restdocs.domain.type

enum class BasicType {
    INT,
    LONG,
    FLOAT,
    DOUBLE,
    STRING,
    BOOLEAN;

    fun asDataType(): DataType {
        return when (this) {
            INT -> DataType.INT
            LONG -> DataType.LONG
            FLOAT -> DataType.FLOAT
            DOUBLE -> DataType.DOUBLE
            STRING -> DataType.STRING
            BOOLEAN -> DataType.BOOLEAN
        }
        throw UnsupportedOperationException("Could not map $this to DataType")
    }
}
